test/bdd/pkg/edv/edv_steps.go

```diff
@@ -49,71 +73,325 @@ func (e *Steps) createDataVault(vaultID, expectedVaultLocation string) error {
+func (e *Steps) decryptDocument() error {
+	decryptedEnvelope, err := e.bddContext.Packer.Unpack(e.bddContext.ReceivedEncryptedDoc.JWE)
+	if err != nil {
+		return err
+	}
+
+	decryptedDoc := operation.StructuredDocument{}
+
+	err = json.Unmarshal(decryptedEnvelope.Message, &decryptedDoc)
+	if err != nil {
+		return err
+	}
+
+	err = verifyStructuredDocsAreEqual(decryptedDoc, e.bddContext.StructuredDocToBeEncrypted)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (e *Steps) receiveDocument(docID, vaultID, expectedDocumentArgKey string) error {
-	client := edv.New(e.bddContext.EDVHostURL)
+func verifyEncryptedDocsAreEqual(retrievedDocument, expectedDocument operation.EncryptedDocument) error {
+	if retrievedDocument.ID != expectedDocument.ID {
+		return unexpectedValueError(expectedDocument.ID, retrievedDocument.ID)
+	}
+
+	if retrievedDocument.Sequence != expectedDocument.Sequence {
+		return unexpectedValueError(string(expectedDocument.Sequence), string(retrievedDocument.Sequence))
+	}
 
-	document, err := client.ReadDocument(vaultID, docID)
+	err := verifyJWEFieldsAreEqual(expectedDocument, retrievedDocument)
 	if err != nil {
 		return err
 	}
 
-	expectedDocument := e.bddContext.Args[expectedDocumentArgKey]
+	return nil
+}
 
-	if string(document) != expectedDocument {
-		return fmt.Errorf("expected the document to be %s, got %s instead", expectedDocument, string(document))
+func verifyJWEFieldsAreEqual(expectedDocument, retrievedDocument operation.EncryptedDocument) error {
+	// CouchDB likes to switch around the order of the fields in the JSON.
+	// This means that we can't do a direct string comparison of the JWE (json.rawmessage) fields
+	// in the EncryptedDocument structs. Instead we need to check each field manually.
+	var expectedJWEFields map[string]string
+
+	err := json.Unmarshal(expectedDocument.JWE, &expectedJWEFields)
+	if err != nil {
+		return err
+	}
+
+	expectedProtectedFieldValue, expectedIVFieldValue, expectedCiphertextFieldValue, expectedTagFieldValue,
+		err := getJWEFieldValues(expectedJWEFields, "expected JWE")
+	if err != nil {
+		return err
+	}
+
+	var retrievedJWEFields map[string]string
+
+	err = json.Unmarshal(retrievedDocument.JWE, &retrievedJWEFields)
+	if err != nil {
+		return err
+	}
+
+	retrievedProtectedFieldValue, retrievedIVFieldValue, retrievedCiphertextFieldValue, retrievedTagFieldValue,
+		err := getJWEFieldValues(retrievedJWEFields, "retrieved JWE")
+	if err != nil {
+		return err
+	}
+
+	err = verifyFieldsAreEqual(
+		retrievedProtectedFieldValue, expectedProtectedFieldValue,
+		retrievedIVFieldValue, expectedIVFieldValue,
+		retrievedCiphertextFieldValue, expectedCiphertextFieldValue,
+		retrievedTagFieldValue, expectedTagFieldValue)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getJWEFieldValues(jweFields map[string]string, jweDocType string) (string, string, string, string, error) {
+	protectedFieldValue, found := jweFields["protected"]
```

Comment:
create var/const for map keys used in this function

Reply:
Done